tests/test_get_describe_only_query_result.c

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2018-2020 Snowflake Computing, Inc. All rights reserved.
+ */
+#include "utils/test_setup.h"
+#include <cJSON.h>
+#include <string.h>
+#include <connection.h>
+#include <memory.h>
+#include <error.h>
+
+/**
+ * Tests fetching the describe only query result response directly from SFSTMT
+ * @param unused
+ */
+void test_get_describe_only_query_result(void **unused) {
+    SF_CONNECT *sf = setup_snowflake_connection();
+    SF_STATUS status = snowflake_connect(sf);
+
+    if (status != SF_STATUS_SUCCESS) {
+        dump_error(&(sf->error));
+    }
+    assert_int_equal(status, SF_STATUS_SUCCESS);
+
+    /* query */
+    SF_STMT *sfstmt = snowflake_stmt(sf);
+    SF_QUERY_RESULT_CAPTURE *result_capture;
+    snowflake_query_result_capture_init(&result_capture);
+
+    clear_snowflake_error(&sfstmt->error);
+    status = snowflake_prepare(sfstmt, "select randstr(100,random()) from table(generator(rowcount=>2))", 0);
+    assert_int_equal(status, SF_STATUS_SUCCESS);
+
+    // Issue the query
+    clear_snowflake_error(&sfstmt->error);
+    status = snowflake_describe_with_capture(sfstmt, result_capture);
+    assert_int_equal(status, SF_STATUS_SUCCESS);
+    char *resultBuffer = result_capture->capture_buffer;
+
+    // Parse the JSON, and grab a few values to verify correctness
+    cJSON *parsedJSON = snowflake_cJSON_Parse(resultBuffer);
+
+    sf_bool success;
+    json_copy_bool(&success, parsedJSON, "success");
+    assert_int_equal(success, SF_BOOLEAN_TRUE);
+
+    cJSON *data = snowflake_cJSON_GetObjectItem(
+            parsedJSON, "data");
+    assert_int_equal(
+            strlen(snowflake_cJSON_GetObjectItem(data, "queryID")->valuestring) + 1,
+            SF_UUID4_LEN);
+
+    // Make sure that the query is run in describe only mode and the actual result is empty
```

Comment:
Could you add a few more asserts to check that what we need in Snowpark prepare is in this response? I.e., column metadata info.

Reply:
ack